pkg/ansible/watches/watches.go

```diff
@@ -104,65 +101,65 @@ func parseLabelSelector(dls tempLabelSelector) metav1.LabelSelector {
-	Values   []string                     `yaml:"values,omitempty"`
+	Key      string                       `json:"key"`
+	Operator metav1.LabelSelectorOperator `json:"operator"`
+	Values   []string                     `json:"values,omitempty"`
 }
 
-// UnmarshalYAML - implements the yaml.Unmarshaler interface for Watch.
-// This makes it possible to verify, when loading, that the GroupVersionKind
-// specified for a given watch is valid as well as provide sensible defaults
-// for values that were omitted.
-func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	// Use an alias struct to handle complex types
-
-	type alias struct {
-		Group                       string                    `yaml:"group"`
-		Version                     string                    `yaml:"version"`
-		Kind                        string                    `yaml:"kind"`
-		Playbook                    string                    `yaml:"playbook"`
-		Role                        string                    `yaml:"role"`
-		Vars                        map[string]interface{}    `yaml:"vars"`
-		MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
-		ReconcilePeriod             string                    `yaml:"reconcilePeriod"`
-		ManageStatus                bool                      `yaml:"manageStatus"`
-		WatchDependentResources     bool                      `yaml:"watchDependentResources"`
-		WatchClusterScopedResources bool                      `yaml:"watchClusterScopedResources"`
-		Blacklist                   []schema.GroupVersionKind `yaml:"blacklist"`
-		Finalizer                   *Finalizer                `yaml:"finalizer"`
-		Selector                    tempLabelSelector         `yaml:"selector"`
-	}
-	var tmp alias
+// Use an alias struct to handle complex types
+type alias struct {
+	Group                       string                    `yaml:"group"`
+	Version                     string                    `yaml:"version"`
+	Kind                        string                    `yaml:"kind"`
+	Playbook                    string                    `yaml:"playbook"`
+	Role                        string                    `yaml:"role"`
+	Vars                        map[string]interface{}    `yaml:"vars"`
+	MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
+	ReconcilePeriod             *metav1.Duration          `yaml:"reconcilePeriod,omitempty"`
+	ManageStatus                *bool                     `yaml:"manageStatus,omitempty"`
+	WatchDependentResources     *bool                     `yaml:"watchDependentResources,omitempty"`
+	WatchClusterScopedResources *bool                     `yaml:"watchClusterScopedResources,omitempty"`
+	Blacklist                   []schema.GroupVersionKind `yaml:"blacklist,omitempty"`
+	Finalizer                   *Finalizer                `yaml:"finalizer"`
+	Selector                    tempLabelSelector         `yaml:"selector"`
```

Comment:
We have not tests to ensure the`tempLabelSelector`, wdyt about add it as well. The goal is to ensure that it will have the values parsed from the file. So, for this scenario and for the Blacklist we need to ensure that when the value exists in the watch.yaml it will be parsed and when it is not informed the default one will be used.